app/javascript/mastodon/components/short_number.tsx

```diff
@@ -0,0 +1,111 @@
+import { memo } from 'react';
+
+import { FormattedMessage, FormattedNumber } from 'react-intl';
+
+import { toShortNumber, pluralReady, DECIMAL_UNITS } from '../utils/numbers';
+
+/**
+ * Custom renderer for numbers
+ * @param {JSX.Element} displayNumber Number to display
+ * @param {number} pluralReady Number used for pluralization
+ * @returns {JSX.Element} Final render of number
+ */
+interface ShortNumberRenderer {
+  (displayNumber: JSX.Element, pluralReady: number): JSX.Element;
+}
+
+/**
+ * Component that renders short big number to a shorter version
+ * @param {ShortNumberProps} props  Props for the component
+ * @param {number} props.value value Number to display in short variant
+ * @param {ShortNumberRenderer} [props.renderer]
+ * Custom renderer for numbers, provided as a prop. If another renderer
+ * passed as a child of this component, this prop won't be used.
+ * @param {ShortNumberRenderer} [props.children]
+ * Custom renderer for numbers, provided as a child. If another renderer
+ * passed as a prop of this component, this one will be used instead.
+ * @returns {JSX.Element} - Rendered number.
+ */
+interface ShortNumberProps {
+  value: number;
+  renderer?: ShortNumberRenderer;
+  children?: ShortNumberRenderer;
+}
+export const ShortNumberRenderer: React.FC<ShortNumberProps> = ({
+  value,
+  renderer,
+  children,
+}): JSX.Element => {
+  const shortNumber = toShortNumber(value);
+  const [, division] = shortNumber;
+
+  if (children != null && renderer != null) {
+    console.warn(
+      'Both renderer prop and renderer as a child provided. This is a mistake and you really should fix that. Only renderer passed as a child will be used.'
+    );
+  }
+
+  const customRenderer = children != null ? children : renderer || null;
+
+  const displayNumber = <ShortNumberCounter value={shortNumber} />;
+
+  return customRenderer != null
```

Comment:
You should be able to use new Typescript syntax like this:

```ts
return customRenderer?(displayNumber, …) || displayNumber
```

Reply:
Is this optional chaining?
It seems to be interpreted as a conditional operator by eslint and results in an error.

I tried to match the above
````
  return customRenderer
    ? customRenderer(displayNumber, pluralReady(value, division))
    : displayNumber;.
```

Reply:
Yes, this is optional chaining.

Sorry for my first comment, I think the correct syntax should be `return customRenderer?.(displayNumber, …) || displayNumber`

Reply:
`customRenderer? (displayNumber, ...)` also passed eslint.
I didn't know how to write functions with optional chaining, that's good to know ✌️